### ISSUE TITLE
Add "Form" support for the OpenAPI v2/v3 'readOnly/writeOnly' JSON schema attributes

### ISSUE THREAD

**Issue by rossmcdonald** [stoplightio/studio#58]
**User story.**
As an API developer, I want to be able to mark certain items as readOnly/writeOnly.

**Describe the solution you'd like**
Form view elements should be added for setting readOnly/writeOnly attributes:

- OpenAPI v2 only supports `readOnly`
- OpenAPI v3 supports both `readOnly` and `writeOnly`

**Comment by philsturgeon** [comment]
For our JSON Schema stand-alone users, they can use readOnly/writeOnly if they’re editing draft 07 but not before.

**Comment by thomasruiz** [comment]
This should be a higher priority. `readOnly` and `writeOnly` is a must have for reusability, which is very important with OpenAPI.

**Comment by philsturgeon** [comment]
@thomasruiz I hear you, but we are having to use our priorities conservatively. We have a lot of functionality to get done, and plenty of users want plenty of things. For now, adding a `readOnly: true` or `writeOnly: true` can be done in the Write panel, which should unblock you. I hope to see this soon but there are many other features coming to be excited about too.

**Comment by thomasruiz** [comment]
@philsturgeon it works for writing the specs, but the docs won't reflect that, and  the "try it" tab. Problem half solved, I guess. I understand that you have other priorities, a lot of upcoming features seem promising! I'll just use something else to generate and host docs for now.

**Comment by philsturgeon** [comment]
@thomasruiz ahh, well Docs is a different thing to the Form view which is what this issue is about. 

@lottamus do we have any issues which talk about getting readOnly and writeOnly to docs?

**Comment by thomasruiz** [comment]
@philsturgeon I'm going to create another issue for this specific problem.

**Comment by greemo** [comment]
+1 for readOnly/writeOnly in the properties dialog

**Comment by jensljungblad** [comment]
@philsturgeon Did you ever add support for readOnly to the docs? Read only attributes show up in the docs for `POST`/`PUT` etc, which is pretty confusing!

**Comment by philsturgeon** [comment]
@jensljungblad It was put in the icebox for a little while but it's being handled as part of some major improvements to docs which will hopefully be out before the end of the year. Fingers crossed!

**Comment by jensljungblad** [comment]
> @jensljungblad It was put in the icebox for a little while but it's being handled as part of some major improvements to docs which will hopefully be out before the end of the year. Fingers crossed!

That's great to hear!🤞

**Comment by philsturgeon** [comment]
Alright, readOnly properties in docs will no longer show up for writes, and vice versa. This will be deployed pretty soon, and I'm moving this issue to our internal tracker so we can get Form support worked on too. 

Sorry this one took a while, but you'll see it all working as you expect really soon. Cheers!
